src/backend/core/models/workflow_metrics_models.py

```
@@ -0,0 +1,362 @@
+    
+    def track_keyword_search(self, latency_ms: float, returned_keywords: List[str]) -> None:
+        """Track keyword search performance."""
+        self.keyword_search_stats["calls"] += 1
+        self.keyword_search_stats["total_latency_ms"] += latency_ms
+        self.keyword_search_stats["avg_latency_ms"] = (
+            self.keyword_search_stats["total_latency_ms"] / 
+            self.keyword_search_stats["calls"]
+        )
+        self.keyword_search_stats["returned_keywords"].extend(returned_keywords)
+    
+    def track_pattern_learning(self, predicted: bool, keyword_count: int, accuracy: float = 0.0) -> None:
```

Ruff (0.14.10)

Boolean-typed positional argument in function definition

(FBT001)